### ISSUE TITLE
EnableUnsafeBinaryFormatterSerialization should be off by default for all projects for .NET 9+

### ISSUE THREAD

**Issue by bartonjs** [dotnet/sdk#42311]
https://github.com/dotnet/sdk/blob/67b96df9997212e78c15ffc0813f76be19be40cd/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L161-L198 enables EnableUnsafeBinaryFormatterSerialization for Winforms and WPF workloads by default.

For .NET 9 we have changed resource deserialization to no longer use BinaryFormatter, so starting with the .NET 9 TFM this property should be off for all project types, requiring explicit opt-in whenever someone wants it.

**Comment by jeffhandley** [comment]
I think we can remove that entire block of logic and comments, never setting a default for `EnableUnsafeBinaryFormatterSerialization`. But there are related aspects I'm unsure of.

Where we set `GenerateResourceWarnOnBinaryFormatterUse` to `true`:
https://github.com/dotnet/sdk/blob/f6b5eb1c6031d05627f62c2a052dea206fd56ea6/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L106-L108

Where MSBuild defines the `GenerateResource.BinaryFormatterUse` warning string (and then also the places where this is consumed):
https://github.com/dotnet/msbuild/blob/858aa179d0a8b8101ee41d001966ce8745b3e680/src/Tasks/Resources/Strings.resx#L1164-L1168

**Comment by bartonjs** [comment]
I don't know we can get rid of it entirely... presumably when using the net9 SDK to build a net8 project we still want it on for WF/WPF?

**Comment by lonitra** [comment]
Taking a look again is there any action that needs to be done here? It looks like this is only enabled for .NET 8 and below for WinForms and WPF:
https://github.com/dotnet/sdk/blob/67b96df9997212e78c15ffc0813f76be19be40cd/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L191-L192

**Comment by bartonjs** [comment]
Oh, wow, it does say "Less", not "Greater".  Um.... yeah, might be "already done"

**Comment by lonitra** [comment]
I checked this in WinForms + WPF app that attempts to use BinaryFormatter to serialize, and I see that after pulling in BinaryFormatter package and trying to run the app I get exception that BinaryFormatter is disabled. So yes it looks like `EnableUnsafeBinaryFormatterSerialization` is no longer automatically enabled in .NET 9 for WinForms + WPF. 
I had also tried "downgrading" those same .NET 9 apps to .NET 8 by changing the TargetFramework and running it and I get no exception about BinaryFormatter being disabled.
